manuscript/3-SoftConstr.txt

```diff
@@ -81,33 +80,33 @@
   * Επαναχρησιμοποίηση
   * Πρότυπα στην κατασκευή
 
-Οι τέσσερις πρώτες αρχές ισχύουν τόσο για τη σχεδίαση όσο και για
+Οι πρώτες τέσσερις αρχές ισχύουν τόσο για τη σχεδίαση όσο και για
 την κατασκευή.
-Οι επόμενες παράγραφοι ορίζουν τις αρχές αυτές και περιγράφουν πώς
-ισχύουν για την κατασκευή λογισμικού.
+Οι επόμενες παράγραφοι ορίζουν αυτές τις αρχές και περιγράφουν πώς
+εφαρμόζονται στην κατασκευή λογισμικού.
 
 ### Ελαχιστοποίηση περιπλοκότητας
 
 
 Οι περισσότεροι άνθρωποι δυσκολεύονται να συγκρατήσουν περίπλοκες
 δομές και πληροφορίες στην λειτουργική μνήμη τους, ειδικά σε μεγάλα
 χρονικά διαστήματα. Αυτό αποδεικνύεται ότι είναι ένας σημαντικός παράγοντας
-που επηρεάζει πώς οι άνθρωποι μεταφέρουν την συμπεριφορά αυτή στους
+που επηρεάζει πώς οι άνθρωποι μεταφέρουν την αυτή συμπεριφορά στους
 υπολογιστές και το γεγονός αυτό οδηγεί σε μια από τις πιο ισχυρές
-πορείες στην κατασκευή λογισμικού, την ελαχιστοποίηση της Πολυπλοκότητας.
+πορείες στην κατασκευή λογισμικού, την ελαχιστοποίηση της πολυπλοκότητας.
 Η ανάγκη για μείωση της περιπλοκότητας αντικατοπτρίζεται ουσιαστικά
 σε κάθε πτυχή της κατασκευής λογισμικού και είναι ιδιαίτερα κρίσιμη
 στον έλεγχο της κατασκευής λογισμικών.
 
```

Comment:
Several types of complexity can affect
software construction. Tools can be used to
manage different aspects of the complexity of
software components and their construction.
For example, cyclomatic complexity is a
static analysis measure of how difficult code
is to test and understand. The tool, developed
by Thomas J. McCabe, Sr., in 1976,
calculates the number of linearly independent
paths through a program’s source code.
Ideally, there should be at least that number
of test cases. Other examples are tools like
Make, which can build an application, or
integrated development environments (IDEs)
for entering, editing and compiling code.
These tools help manage the complexity of
the construction process.

Δε μπορώ να βρω αυτό το κομμάτι!